src/relation.js

```diff
@@ -179,7 +179,7 @@ export default RelationBase.extend({
       knex.join(
         joinTable,
         joinTable + '.' + targetKey, '=',
-        this.targetTableName + '.' + this.targetIdAttribute
+        this.targetTableName + '.' + (this.otherKeyTarget || this.targetIdAttribute)
```

Comment:
This also is not self describing why it is done. Again `isTargetedForeignKey()` would be recommended way to determine which key to use.